modules/nf-core/nextdenovo/main.nf

```diff
@@ -0,0 +1,94 @@
+process NEXTDENOVO {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda "bioconda::nextdenovo=2.5.2"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/nextdenovo%3A2.5.2--py39h20169af_0':
+        'biocontainers/nextdenovo:2.5.2--py39h20169af_0' }"
+
+    input:
+    tuple val(meta), path(reads)
+    val (read_type)   // clr, ont, hifi
+    val (step) 	      // all, correct, assemble
+    val (input_type)  // raw, corrected
+    val (genome_size) // estimated genome size, suffix K/M/G recognized
+
+    output:
+    tuple val(meta), path("*cns.fasta.gz")  , emit: corrected_reads
+    tuple val(meta), path("*asm.fasta.gz")  , emit: assembly
+    tuple val(meta), path("*asm.fasta.stat"), emit: stat
+    path "versions.yml"                     , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args_general = task.ext.args_general ?: ''
```

Comment:
Hmmm. The general consensus is that we should only use `args`, `args2`, `args3` etc. Personally I feel this is much more readable like this, but it makes the whole set of modules harder to maintain when there are lots of different named `task.ext.X` values.